// @dargmuesli/nuxt-cookie-control@9.1.10/dist/runtime/components/CookieControl.d.vue.ts
import type { Locale } from '#cookie-control/types';
type __VLS_Props = {
    locale?: Locale;
};
declare var __VLS_14: {}, __VLS_16: {}, __VLS_24: {}, __VLS_26: {
    cookie: any;
};
type __VLS_Slots = {} & {
    bar?: (props: typeof __VLS_14) => any;
} & {
    controlButton?: (props: typeof __VLS_16) => any;
} & {
    modal?: (props: typeof __VLS_24) => any;
} & {
    cookie?: (props: typeof __VLS_26) => any;
};
declare const __VLS_base: import("vue").DefineComponent<__VLS_Props, {
    accept: () => void;
    acceptPartial: () => void;
    decline: () => void;
}, {}, {}, {}, import("vue").ComponentOptionsMixin, import("vue").ComponentOptionsMixin, {}, string, import("vue").PublicProps, Readonly<__VLS_Props> & Readonly<{}>, {}, {}, {}, {}, string, import("vue").ComponentProvideOptions, false, {}, any>;
declare const __VLS_export: __VLS_WithSlots<typeof __VLS_base, __VLS_Slots>;
declare const _default: typeof __VLS_export;
export default _default;
type __VLS_WithSlots<T, S> = T & {
    new (): {
        $slots: S;
    };
};
